src/java/org/apache/cassandra/service/StorageProxy.java

```diff
@@ -2744,6 +2744,7 @@ public static Future<Void> submitHint(Mutation mutation,
                                           EndpointsForToken targets,
                                           AbstractWriteResponseHandler<IMutation> responseHandler)
     {
+        targets = targets.filter(StorageProxy::shouldHint);
```

Comment:
if `targets` are "empty" (whatever that means), we might return just here so we would not `submitHit` at all